### PR TITLE
Ah gabw 1708: Orch changes to support catalog permissions

### PR DESCRIPTION
**Pull request by ahaessly**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project/Dependencies.scala
Comment (davidangb) [inline project/Dependencies.scala]: once your rawls PR merges, this will change, right?

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala]: always returns true?

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala]: the three methods below this are not used anywhere in the code, according to IntelliJ. Do we still need them?

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala]: remove comments

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala:56]: I think you should keep the `asAdmin` and `asCurator` pattern going here, and add something like:
```
  def asPermitted(ns: String, name: String, lvl: WorkspaceAccessLevel, userInfo: UserInfo)(op: => Future[PerRequestMessage]): Future[PerRequestMessage] = {
    hasAccessOrAdmin(ns, name, lvl, userInfo) flatMap { isPermitted =>
      if (isPermitted) op else Future.failed(new FireCloudExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, s"You must have at least ${lvl.toString} access.")))
    }
  }
```

if you have that, then you can simplify the methods in WorkspaceService, for example:
```
  def getCatalog(workspaceNamespace: String, workspaceName: String, userInfo: UserInfo): Future[PerRequestMessage] = {
    asPermitted(workspaceNamespace, workspaceName, WorkspaceAccessLevels.Read, userInfo) {
      rawlsDAO.getCatalog(workspaceNamespace, workspaceName) map (RequestComplete(_))
    }
  }
```

disclaimer: I haven't actually tried this outside of unit tests.

File: src/main/resources/swagger/api-docs.yaml
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: "workspace ACLs" or "catalog permissions" (or something else)?

File: src/main/resources/swagger/api-docs.yaml
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: description seems wrong

File: src/main/resources/swagger/api-docs.yaml
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: "workspace ACLs" or "catalog permissions" (or something else)?

File: src/main/resources/swagger/api-docs.yaml
Comment (davidangb) [inline src/main/resources/swagger/api-docs.yaml]: catalog permissions

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala]: see comment below about an `asPermitted` wrapper. If you *don't* implement `asPermitted`, this should be `Future.failed` (same for getCatalog).

**Review by davidangb — CHANGES_REQUESTED** [review]
comments inline!

File: project/Dependencies.scala
Comment (ahaessly) [reply project/Dependencies.scala]: yes

**Review by davidangb — COMMENTED** [review]
looking good! Just need to hold this until rawls merges so you can get the version number for the model jar.

File: src/main/resources/swagger/api-docs.yaml
Comment (rushtong) [inline src/main/resources/swagger/api-docs.yaml]: Does the 403 leak too much information? We don't return a 403 for regular workspace calls and treat them as 404's. Also applies to other calls as well.

**Review by rushtong — APPROVED** [review]
👍  Pending the answer to 403s.

File: src/main/resources/swagger/api-docs.yaml
Comment (ahaessly) [reply src/main/resources/swagger/api-docs.yaml]: Yes, for the get we will return a 404, but for the patch, we will return a 403 if they have read or write but not owner or admin.

**Review by davidangb — APPROVED** [review]

**Comment by davidangb** [comment]
:+1: as soon as tests pass!

**Pull request merged**
